database.go

```diff
@@ -59,6 +59,27 @@ func isKnownArch(arch string) bool {
 	return false
 }
 
+// normalizeOSAndVersion splits the provided platform specifier OS segment
+// into an OS and OSVersion.
+// The expected format is `<OS>[(<OSVersion>)]`, e.g., `windows(10.0.17763)`.
+// If `<os>` is not provided, the current host OS is returned as the first value.
+// If optional `(<OSVersion>)` is not provided, an empty string is returned as the
+// second value.
+func normalizeOSAndVersion(OSAndVersion string) (OS string, OSVersion string) {
+	if OSAndVersion == "" {
+		return runtime.GOOS, ""
+	}
+
+	parts := osVersionRe.Split(OSAndVersion, -1)
```

Comment:
Not critical, but we should benchmark; this was the part that was not needed with the regex in https://github.com/kiashok/platforms/pull/1 (as it already did matched the regular expression, so we could consume the results of it directly, without having to execute another regular expression);

```go
// First element is <os>[(<OSVersion>)]
osVer := osAndVersionRe.FindStringSubmatch(part)
if osVer == nil {
	return specs.Platform{}, fmt.Errorf("%q is an invalid OS component of %q: OSAndVersion specifier component must match %q: %w", part, specifier, osAndVersionRe.String(), errInvalidArgument)
}
p.OS = normalizeOS(osVer[1])
p.OSVersion = osVer[2]
```

Reply:
@thaJeztah  like I mentioned here https://github.com/containerd/platforms/pull/6#discussion_r1556129289 the regex you had in your changes does not pass the tests I added.

Reply:
Curious; do you know which test didn't pass? I rebased my PR and ran the tests and they all were green.

Reply:
![image](https://github.com/containerd/platforms/assets/99994218/45e8396a-ddf2-469f-a201-11ffc1f2e37a)

the three added here. platform_test.go in this PR

Reply:
I opened https://github.com/containerd/platforms/pull/9 to test in CI, but it's passing.

Reply:
@thaJeztah sorry for the confusion, I figured where the issue was when I was testing PR with the regex you shared. Pushed new changes.